tests/pkcs11_test.c

```diff
@@ -130,38 +263,418 @@ static int s_test_pkcs11_lib_omit_initialize(struct aws_allocator *allocator, vo
+    ASSERT_SUCCESS(aws_pkcs11_lib_open_session(pkcs11_lib, created_slot, &session_to_create_key_2 /*out*/));
+
+    /* Login user */
+    ASSERT_SUCCESS(aws_pkcs11_lib_login_user(pkcs11_lib, session_to_access_key, user_pin));
+
+    unsigned long created_key_1 = s_pkcs11_create_key(pkcs11_lib,
+                                                      created_slot,
+                                                      key_label_1,
+                                                      key_id_1,
+                                                      session_to_create_key_1);
+    unsigned long created_key_2 = s_pkcs11_create_key(pkcs11_lib,
+                                                      created_slot,
+                                                      key_label_2,
+                                                      key_id_2,
+                                                      session_to_create_key_2);
+
+    /* Since there are 2 keys, a lookup without label should fail */
+    struct aws_string* key_label_2_str = aws_string_new_from_c_str(allocator, key_label_2);
+    pkey_handle = ULONG_MAX;
+    ASSERT_FAILS(
+        aws_pkcs11_lib_find_private_key(pkcs11_lib,
+                                        session_to_access_key,
+                                        NULL,
+                                        &pkey_handle));
+
+    /* a lookup with label for the first key should find the first key */
+    pkey_handle = ULONG_MAX;
+    ASSERT_SUCCESS(
+        aws_pkcs11_lib_find_private_key(pkcs11_lib,
+                                        session_to_access_key,
+                                        key_label_str,
+                                        &pkey_handle));
+    ASSERT_TRUE(pkey_handle == created_key_1);
+
+    /* a lookup with label for the second key should find the second key */
+    pkey_handle = ULONG_MAX;
     ASSERT_SUCCESS(
-        aws_pkcs11_lib_find_private_key(pkcs11_lib, session_handle, s_pkcs11_tester.pkey_label, &pkey_handle));
+        aws_pkcs11_lib_find_private_key(pkcs11_lib,
+                                        session_to_access_key,
+                                        key_label_2_str,
+                                        &pkey_handle));
+    ASSERT_TRUE(pkey_handle == created_key_2);
 
     /* Clean up */
-    aws_pkcs11_lib_close_session(pkcs11_lib, session_handle);
+    aws_string_destroy(key_label_str);
+    aws_string_destroy(key_label_2_str);
+    aws_string_destroy(user_pin);
+    aws_pkcs11_lib_close_session(pkcs11_lib, session_to_access_key);
+    aws_pkcs11_lib_close_session(pkcs11_lib, session_to_create_key_1);
+    aws_pkcs11_lib_close_session(pkcs11_lib, session_to_create_key_2);
+    s_pkcs11_clear_softhsm(pkcs11_lib);
     aws_pkcs11_lib_release(pkcs11_lib);
     s_pkcs11_tester_clean_up();
     return AWS_OP_SUCCESS;
 }
 AWS_TEST_CASE(pkcs11_find_private_key, s_test_pkcs11_find_private_key)
+
```

Comment:
if you can break tests up at all, please do so

Reply:
Will do that